keps/sig-node/xxx-dynamic-resource-allocation/README.md

```diff
@@ -0,0 +1,1368 @@
+    //
+    // This field is read-only and no changes will be made by Kubernetes
+    // to the ResourceClaim after it has been created.
+   Template *ResourceClaimTemplate
+}
+
+type ResourceClaimTemplate struct {
+    // May contain labels and annotations that will be copied into the ResourceClaim
+    // when creating it. No other fields are allowed and will be rejected during
+    // validation.
+    metav1.ObjectMeta
+
+    // The specification for the ResourceClaim. The entire content is
+    // copied unchanged into the ResourceClaim that gets created from this
+    // template. The same fields as in a ResourceClaim
+    // are also valid here.
+    Spec ResourceClaimSpec
+}
+```
+
+
+Several components must be implemented or modified:
+- A new controller in kube-controller-manager which generated
+  ResourceClaims from Pod ResourceClaimTemplates, similar to
+  https://github.com/kubernetes/kubernetes/tree/master/pkg/controller/volume/ephemeral
+- A kube-scheduler plugin must detect Pods which reference a
+  ResourceClaim (directly or through a template) and ensure that the
+  resource is allocated before the Pod gets scheduled, similar to
+  https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/scheduling/scheduler_binder.go
+- Some utility library similar to
+  https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner
+  and the code in driver-registrar: this will be developed outside of
+  Kubernetes and does not have to be used by plugins, therefore it is
+  not described further in this KEP.
+- Kubelet must be extended to retrieve information from ResourceClaims
+  and then invoke local plugin methods. It must pass information about
+  the additional resources to the container runtime. It must detect
+  whether the container runtime has the necessary support and
+  advertise that in the cluster via node labels to simplify deployment
+  of plugins and Pod scheduling.
+
+```
+<<[UNRESOLVED @pohly]>>
+All of the changes in Kubernetes need to be specified in a lot more detail.
+
+A flow diagram showing the state transitions of a ResourceClaim needs to be added.
+
+Upgrade and downgrade scenarios should already be considered for v1alpha1 to ensure
+that whatever changes will be needed are in place before going to v1beta1 where
+downgrades have to be supported.
+
+All of that will be added once there is consensus to move ahead with this proposal.
+<<[/UNRESOLVED]>>
+```
+
+The gRPC interface is inspired by
+[CSI](https://github.com/container-storage-interface/spec/blob/master/spec.md),
+with “volume” replaced by “resource” and volume specific parts
+removed.
+
+```
+<<[UNRESOLVED @pohly]>>
+Do plugin operations need secrets? They are currently not part of the proposed Kubernetes API.
+<<[/UNRESOLVED]>>
+```
+
+### Node Service RPC
+
+#### `NodePrepareResource`
+
+This RPC is called by kubelet when a Pod that wants to use the
+specified resource is scheduled on a node.  The Plugin SHALL assume
```

Comment:
So this plugin interface with kubelet and generate a CDI format file. Then the runtime, e.g. containerd/CRI-O, will consume the CDI file and translate it to OCI format which is processed by low level runtime, such as kata container?

Reply:
Correct.

That's where the CDI dependency of this KEP comes from.